### PR TITLE
Sbo and cvterms

### PR DESCRIPTION
**Pull request by longxf21**
finished coding

### CODE REVIEW

**Comment by longxf21** [comment]
I also enabled multiple annotations show in hover

**Comment by longxf21** [comment]
When merging sbo terms and visual indication for annotation, need to support visual indication for sbo terms.

File: vscode-antimony/src/server/stibium/stibium/antimony.lark [hunk 1]
```diff
@@ -12,12 +12,44 @@ species : [NUMBER] ["$"] NAME
...
+    | "encodement"
+    | "occursIn"
+    | "container"
+    | "hasProperty"
+    | "propertyBearer"
+    | "property"
+    | "isPropertyOf"
+    | "hasTaxon"
+    | "taxon"
+
```
Comment (mastevb) [inline vscode-antimony/src/server/stibium/stibium/antimony.lark:50]: 2 empty lines.

**Review by mastevb — CHANGES_REQUESTED** [review]

File: vscode-antimony/src/server/stibium/stibium/analysis.py [hunk 2]
```diff
@@ -430,6 +440,19 @@ def handle_annotation(self, scope: AbstractScope, annotation: Annotation):
         self.table.insert(qname, SymbolType.Parameter)
         self.table.insert_annotation(qname, annotation)
     
+    def pre_handle_sboterm(self, scope: AbstractScope, sboterm: Sboterm):
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/analysis.py:443]: Is this method necessary if we're just appending to a list?

File: vscode-antimony/src/server/stibium/stibium/analysis.py [hunk 3]
```diff
@@ -430,6 +440,19 @@ def handle_annotation(self, scope: AbstractScope, annotation: Annotation):
         self.table.insert(qname, SymbolType.Parameter)
         self.table.insert_annotation(qname, annotation)
     
+    def pre_handle_sboterm(self, scope: AbstractScope, sboterm: Sboterm):
+        self.pending_sboterms.append((scope, sboterm))
+    
+    def handle_sboterm_list(self):
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/analysis.py:446]: Is method internal to the class? Should start with an "_". If this is external, you should document what the function does. Also, what does "handle" mean? Is there a more descriptive verb?

File: vscode-antimony/src/server/stibium/stibium/analysis.py [hunk 4]
```diff
@@ -654,6 +677,20 @@ def process_is_assignment(self, node, scope):
         var = self.table.get(qname)
         if len(var) == 0:
             self.warning.append(VarNotFound(name.range, name.text))
+            
+    def process_annotation(self, node, scope):
```
Comment (joseph-hellerstein) [inline vscode-antimony/src/server/stibium/stibium/analysis.py:681]: What does process mean? Is there a more descriptive verb?

**Review by joseph-hellerstein — COMMENTED** [review]
I had some questions but saw nothing that looked wrong.

File: vscode-antimony/src/syntax/antimony.tmLanguage.json [hunk 5]
```diff
@@ -119,7 +119,7 @@
         "keywords": {
             "patterns": [
                 {
-                    "match": "\\b(?:identity)\\b",
+                    "match": "\\b(?:identity|biological_entity_is|hasPart|part|isPartOf|parthood|isVersionOf|hypernym|hasVersion|version|isHomologTo|homolog|isDescribedBy|description|isEncodedBy|encoder|encodes|encodement|occursIn|container|hasProperty|property|isPropertyOf|propertyBearer|hasTaxon|taxon|sboTerm)\\b",
```
Comment (luciansmith) [inline vscode-antimony/src/syntax/antimony.tmLanguage.json:122]: This is pretty awkward.  Can you use ANNOT_KEYWORD?  Maybe not in a .json file...

**Review by luciansmith — CHANGES_REQUESTED** [review]
This looks fine to me, after Joe's comments have been addressed.

File: vscode-antimony/src/syntax/antimony.tmLanguage.json (hunk 5 above)
Comment (longxf21) [reply vscode-antimony/src/syntax/antimony.tmLanguage.json:122]: > 

Could you be a little more specific about how should I format this?

Comment (luciansmith) [reply vscode-antimony/src/syntax/antimony.tmLanguage.json:122]: Sadly, I don't know how formatting things in .json files works.  My hope was that you could have one element that was x|y|z|p|d|q and then refer to that element in this 'match' line.  But perhaps that's impossible?

If it is, and if it's possible to add line returns without messing things up, I'd do something like:
     [...]  identity|
            biological_entity_is|
            hasPart|
[...]

If not, as-is works, I guess!

Comment (luciansmith) [reply vscode-antimony/src/syntax/antimony.tmLanguage.json:122]: Sounds like we're leaving it as-is!

**Review by luciansmith — APPROVED** [review]

**Pull request merged**